src/test/java/com/emeq/counter/api/FoodsApiTest.java

```diff
@@ -0,0 +1,178 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import com.emeq.counter.model.Food;
+import com.emeq.counter.repository.FoodRepository;
+
+@ActiveProfiles("test")
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class FoodsApiTest {
+
+    @Value("${server.schema}://${server.host}:${local.server.port}")
+    private String baseUrl;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    private FoodRepository foodRepository;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    private final String[] fruits = { "Apple", "Avocado", "Kiwi", "Lime", "Orange", "Watrmelon" };
+
+    @BeforeEach
+    public void init() {
+        for (String fruit : fruits) {
+            foodRepository.save(createFood(fruit));
+        }
+    }
+
+    @Test
+    public void allFoods() {
+        ArrayNode json = fruitsAsArrayNode();
+        final ResponseEntity<ArrayNode> response = this.restTemplate.getForEntity(endpoint("/foods"), ArrayNode.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response.getBody().toString()).isEqualTo(json.toString());
+    }
+
+    @Test
+    public void getFood() {
+        JsonNode json = mapper.createObjectNode();
+        ((ObjectNode) json).put("id", 1);
+        ((ObjectNode) json).put("name", fruits[0]);
+
+        final ResponseEntity<JsonNode> response = this.restTemplate.getForEntity(endpoint("/foods/1"), JsonNode.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response.getBody().toString()).isEqualTo(json.toString());
+    }
+
+    @Test
+    public void createFood() {
+        JsonNode request = mapper.createObjectNode();
+        ((ObjectNode) request).put("name", "Banana");
+
+        JsonNode json = mapper.createObjectNode();
+        ((ObjectNode) json).put("id", 7);
+        ((ObjectNode) json).put("name", "Banana");
+
+        ResponseEntity<JsonNode> response = this.restTemplate.postForEntity(endpoint("/foods"), json, JsonNode.class);
```

Comment:
`json` field should be `request ` field and we shouldn't send id to POST request.

Reply:
Thank you. I missed this line. I fixed it and the tests were successful.